homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,644 @@
+
+    def __init__(
+        self,
+        fritzbox_tools: FritzBoxTools,
+        device_friendly_name: str,
+        switch_info: SwitchInfo,
+    ) -> None:
+        """Init Fritzbox port switch."""
+        super().__init__(fritzbox_tools, device_friendly_name)
+
+        self._description = switch_info["description"]
+        self._friendly_name = switch_info["friendly_name"]
+        self._icon = switch_info["icon"]
+        self._type = switch_info["type"]
+        self._update: Callable = switch_info["callback_update"]
+        self._switch: Callable = switch_info["callback_switch"]
+
+        self._name = f"{self._friendly_name} {self._description}"
+        self._unique_id = (
+            f"{self._fritzbox_tools.unique_id}-{slugify(self._description)}"
+        )
+
+        self._attributes: dict[str, str] = {}
+        self._is_available = True
+
+        self._attr_is_on = False
+
+    @property
+    def name(self):
+        """Return name."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return name."""
+        return self._icon
+
+    @property
+    def unique_id(self):
+        """Return unique id."""
+        return self._unique_id
+
+    @property
+    def available(self) -> bool:
+        """Return availability."""
+        return self._is_available
+
+    @property
+    def extra_state_attributes(self):
+        """Return device attributes."""
+        return self._attributes
+
+    async def async_update(self):
+        """Update data."""
+        _LOGGER.debug("Updating '%s' (%s) switch state", self.name, self._type)
+        await self._update()
+
+    async def async_turn_on(self, **kwargs) -> None:
+        """Turn on switch."""
+        await self._async_handle_turn_on_off(turn_on=True)
+
+    async def async_turn_off(self, **kwargs) -> None:
+        """Turn off switch."""
+        await self._async_handle_turn_on_off(turn_on=False)
+
+    async def _async_handle_turn_on_off(self, turn_on: bool) -> bool:
+        """Handle switch state change request."""
+        await self._switch(turn_on)
+        self._attr_is_on = turn_on
+        return True
+
+
+class FritzBoxPortSwitch(FritzBoxBaseSwitch, SwitchEntity):
+    """Defines a FRITZ!Box Tools PortForward switch."""
+
+    def __init__(
+        self,
+        fritzbox_tools: FritzBoxTools,
+        device_friendly_name: str,
+        port_mapping: dict,
+        idx: int,
+        connection_type: str,
+    ) -> None:
+        """Init Fritzbox port switch."""
+        self._fritzbox_tools: FritzBoxTools = fritzbox_tools
+
+        self._attributes = {}
+        self.connection_type = connection_type
+        self.port_mapping: dict | None = port_mapping  # dict in the format as it comes from fritzconnection. eg: {'NewRemoteHost': '0.0.0.0', 'NewExternalPort': 22, 'NewProtocol': 'TCP', 'NewInternalPort': 22, 'NewInternalClient': '192.168.178.31', 'NewEnabled': True, 'NewPortMappingDescription': 'Beast SSH ', 'NewLeaseDuration': 0}
```

Comment:
```suggestion
        self.port_mapping: port_mapping  # dict in the format as it comes from fritzconnection. eg: {'NewRemoteHost': '0.0.0.0', 'NewExternalPort': 22, 'NewProtocol': 'TCP', 'NewInternalPort': 22, 'NewInternalClient': '192.168.178.31', 'NewEnabled': True, 'NewPortMappingDescription': 'Beast SSH ', 'NewLeaseDuration': 0}
```

Shouldn't need it here if you type it on the constructor